src/kernel/chainstatemanager_opts.h

```diff
@@ -27,6 +27,7 @@ namespace kernel {
 struct ChainstateManagerOpts {
     const CChainParams& chainparams;
     const std::function<NodeClock::time_point()> adjusted_time_callback{nullptr};
+    bool check_block_index{false};
```

Comment:
Same as above. I think it would be best to add a convenience function for libbitcoinkernel users to init the values here instead of using tristates everywhere.